id/_internal/oidc/ambient.py

```diff
@@ -34,6 +35,8 @@
 _GCP_IDENTITY_REQUEST_URL = "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity"  # noqa
 _GCP_GENERATEIDTOKEN_REQUEST_URL = "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/{}:generateIdToken"  # noqa
 
+_env_var_regex = re.compile(r"[^A-Z0-9_]")
```

Comment:
I think we probably need to handle the edge case that the audience starts with a number, since environment variables can't start with numbers.